src/buildstream/sandbox/_sandboxreapi.py

```diff
@@ -184,8 +184,12 @@ def _fetch_action_result_outputs(self, casremote, action_result):
 
             raise SandboxError("Output directory structure had no digest attached.")
 
-        # Fetch stdout and stderr blobs
-        cascache.fetch_blobs(casremote, [action_result.stdout_digest, action_result.stderr_digest])
+        # Fetch stdout and stderr blobs, if they exist
+        blobs = []
+        for digest in [action_result.stdout_digest, action_result.stderr_digest]:
+            if digest.hash:
+                blobs.append(digest)
+        cascache.fetch_blobs(casremote, blobs)
```

Comment:
I think we can short-circuit this even further

```suggestion
        if blobs:
            cascache.fetch_blobs(casremote, blobs)
```

Reply:
This PR looks reasonable incl. the above suggestion.